src/main/resources/templates/fragments/basic_search_form.html

```diff
@@ -4,10 +4,9 @@
     <div class="form-group row">
         <label class="col-sm-2 col-form-label"></label>
 
-        <div class="input-group mb-3">
-
-	    <input type="text" th:field="*{query}" class="form-control" placeholder="Enter a programming/coding query"
-            aria-describedby="js-course-search-submit" autofocus/>
+        
+        <div class="input-group mb-3">    
+        <input type="text" name="query" class="form-control" placeholder="Ener a programming/coding query" th:value="${previousSearch}" aria-describedby="js-course-search-submit" autofocus/>
```

Comment:
There is a typo on this line (should say "Enter" instead of "Ener")

Reply:
Good catch!  We'll fix it!